src/fields.cpp

```diff
@@ -213,6 +213,44 @@ fields_chunk::~fields_chunk() {
   if (new_s && new_s->refcount-- <= 1) delete new_s; // delete if not shared
 }
 
+static void split_into_tiles(grid_volume gvol, std::vector<grid_volume> *result) {
+  const size_t base_nowned_min = 20000;
+  if (gvol.nowned_min() < base_nowned_min) {
+    result->push_back(gvol);
+    return;
+  }
+
+  int best_split_point;
+  direction best_split_direction;
+  double left_effort_fraction;
+  gvol.find_best_split(2, false /* fragment_cost */, best_split_point, best_split_direction,
+                       left_effort_fraction);
+  grid_volume left_gvol = gvol.split_at_fraction(false, best_split_point, best_split_direction);
+  split_into_tiles(left_gvol, result);
+  grid_volume right_gvol = gvol.split_at_fraction(true, best_split_point, best_split_direction);
+  split_into_tiles(right_gvol, result);
+  return;
+}
+
+// First check that the tile volumes gvs do not intersect and that they add
+// up to the chunk's total grid_volume gv
+static void check_tiles(grid_volume gv, const std::vector<grid_volume> &gvs) {
+  grid_volume vol_intersection;
+  for (size_t i = 0; i < gvs.size(); i++)
+    for (size_t j = i + 1; j < gvs.size(); j++)
+      if (gvs[i].intersect_with(gvs[j], &vol_intersection))
+        meep::abort("gvs[%zu] intersects with gvs[%zu]\n", i, j);
+  size_t sum = 0;
+  for (const auto& sub_gv : gvs) {
+    size_t grid_points = 1;
+    LOOP_OVER_DIRECTIONS(sub_gv.dim, d) { grid_points *= sub_gv.num_direction(d); }
+    sum += grid_points;
```

Comment:
You can just use `sum += sub_gv.ntot();`

Reply:
Unfortunately, we have to manually compute the number of grid points in `sub_gv` and can't use `sub_gv.ntot()` because to use the latter we would first need to call its private function [`grid_volume::update_ntot()`](https://github.com/NanoComp/meep/blob/master/src/meep/vec.hpp#L1038) which is not possible (otherwise the number of grid points stored in `the_ntot` and returned by `ntot()` is *not* up to date and causes `check_tiles` to abort on line 251).

Reply:
Fixed.